_dev/gulp/tasks/js/components.js

```diff
@@ -1,17 +1,23 @@
 'use strict';
 
 module.exports = function (gulp, plugins, settings, handlers) {
-  gulp.task('js:components', function (cb) {
+  gulp.task('js:components', ['js:wrap'],function (cb) {
 
     if (settings.debug) {
       console.log('js:components task...');
     }
 
-    return gulp.src(settings.files.pattern_lab.patterns.js)
+    if (settings.cms === 'drupal') {
+      var source = settings.files.tmp.js.drupal;
+    } else if (settings.cms === 'static') {
+      var source = settings.files.tmp.js.static;
+    }
+
+    return gulp.src(source)
       .pipe(plugins.plumber({errorHandler: handlers.error}))
       // .pipe(plugins.sourcemaps.init({loadMaps: true}))
       .pipe(plugins.uglify())
       // .pipe(plugins.sourcemaps.write('./'))
-      .pipe(gulp.dest(settings.path.theme.templates));
+      .pipe(gulp.dest(settings.path.theme.js));
```

Comment:
Again, `components` are files from Pattern Lab components. The idea was to place them together with TWIG and CSS files, inside `templates` folder in Drupal. Then, you manually create `library` for those components and point onto CSS and JS files. They should NOT be combined together with other files, nor moved to `assets` folder.